### PR TITLE
improv: (core-dvm) clean up DVM contracts in prep for audit

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

This PR makes some small refinements, simplifications, gas savings and cleans the Voting contracts in preparation for the next re-audit.

### CODE REVIEW

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 1]
```diff
@@ -736,60 +756,41 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
     // voting activity the voters balance is updated accordingly. The caller can provide a maxTraversals parameter to
     // limit the number of resolved requests to traverse in this call to bound the gas used. Note each iteration of
     // this function re-uses a fresh slash variable to produce useful logs on the amount a voter is slashed.
-    function _updateAccountSlashingTrackers(address voterAddress, uint64 maxTraversals) internal {
-        VoterStake storage voterStake = voterStakes[voterAddress];
+    function _updateAccountSlashingTrackers(address voter, uint64 maxTraversals) internal {
```
Comment (chrismaree) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:759]: I've further simplified this method by adding two internal methods that we can use now: a) `requestSlashingTrackers` (which we previously had as an external method) which returns slashing trackers and re-uses them here so we dont need to re-call this method/double up the code and b) `getVoterParticipation` which returns an enum for a given voter for their participation.

overall, this decreased the line cound and complexity in this function, while saving a bit of gas.

File: packages/core/contracts/data-verification-mechanism/implementation/Staker.sol [hunk 2]
```diff
@@ -276,8 +276,8 @@ abstract contract Staker is StakerInterface, Ownable, Lockable, MultiCaller {
     }
 
     // Updates an account internal trackers.
-    function _updateTrackers(address voterAddress) internal virtual {
```
Comment (chrismaree) [inline packages/core/contracts/data-verification-mechanism/implementation/Staker.sol:279]: I re-named this variable throughout the contracts for consistency and simplification reasons.

File: packages/core/contracts/data-verification-mechanism/implementation/Staker.sol [hunk 3]
```diff
@@ -286,12 +286,12 @@ abstract contract Staker is StakerInterface, Ownable, Lockable, MultiCaller {
 
     /**
      * @notice Gets the pending stake for a voter for a given round.
-     * @param voterAddress the voter address.
+     * @param voter the voter address.
      * @param roundId round id.
      * @return uint256 amount of the pending stake.
      */
-    function getVoterPendingStake(address voterAddress, uint256 roundId) external view returns (uint256) {
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/Staker.sol:293]: pendingStakes is a `mapping(uint256 => uint256) pendingStakes`. Why do he have a `uint32` here?

https://github.com/UMAprotocol/protocol/blob/d6bb02cb3d9b765d820c7547256c55bcdc1f9822/packages/core/contracts/data-verification-mechanism/implementation/Staker.sol#L26

File: packages/core/contracts/data-verification-mechanism/implementation/Staker.sol [hunk 4]
```diff
@@ -355,15 +355,15 @@ abstract contract Staker is StakerInterface, Ownable, Lockable, MultiCaller {
 
     // This function must be called before any tokens are staked. Update the voter's pending stakes when necessary.
     // The contract that inherits from Staker (e.g. VotingV2) must implement this logic by overriding this function.
-    function _computePendingStakes(address voterAddress, uint256 amount) internal virtual;
+    function _computePendingStakes(address voter, uint256 amount) internal virtual;
 
     // Add a new stake amount to the voter's pending stake for a specific round id.
     function _incrementPendingStake(
-        address voterAddress,
-        uint256 roundId,
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/Staker.sol:363]: Same question here: 
pendingStakes is a `mapping(uint256 => uint256) pendingStakes`, is this correct?

https://github.com/UMAprotocol/protocol/blob/d6bb02cb3d9b765d820c7547256c55bcdc1f9822/packages/core/contracts/data-verification-mechanism/implementation/Staker.sol#L26

File: packages/core/contracts/data-verification-mechanism/implementation/Staker.sol (hunk 3 above)
Comment (chrismaree) [reply packages/core/contracts/data-verification-mechanism/implementation/Staker.sol:293]: good catch. cant be `uint32 => uint256`

File: packages/core/contracts/data-verification-mechanism/implementation/FixedSlashSlashingLibrary.sol [hunk 5]
```diff
@@ -9,58 +9,64 @@ import "../interfaces/SlashingLibraryInterface.sol";
  elaborate slashing algorithms via UMA governance.
  */
 
-contract SlashingLibrary is SlashingLibraryInterface {
+contract FixedSlashSlashingLibrary is SlashingLibraryInterface {
```
Comment (Reinis-FRP) [inline packages/core/contracts/data-verification-mechanism/implementation/FixedSlashSlashingLibrary.sol:12]: `SlashingLibrary` was also referenced in `packages/scripts/src/upgrade-tests/voting2/0_Deploy.ts` that should be updated there as well

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 6]
```diff
@@ -65,24 +73,30 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
+    enum VoteParticipation {
+        DidNotVote, // Voter did not vote.
+        WrongVote, // Voter voted against the resolved price.
+        CorrectVote // Voter voted with the resolved price.
+    }
+
     // Only used as a return value in view methods -- never stored in the contract.
     struct RequestState {
         RequestStatus status;
         uint256 lastVotingRound;
```
Comment (Reinis-FRP) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:85]: shall this also be `uint32` as everywhere else?

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 7]
```diff
@@ -598,7 +606,20 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
+        uint256 requestIndex,
+        uint32 lastVotingRound,
+        address voter
+    ) public view returns (VoteParticipation) {
+        VoteInstance storage voteInstance =
+            priceRequests[resolvedPriceRequestIds[requestIndex]].voteInstances[lastVotingRound];
+        bytes32 revealHash = voteInstance.voteSubmissions[voter].revealHash;
+        if (revealHash == bytes32(0)) return VoteParticipation.DidNotVote;
+        else if (voteInstance.results.wasVoteCorrect(revealHash)) return VoteParticipation.CorrectVote;
+        else return VoteParticipation.WrongVote;
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:622]:
```suggestion
       if (voteInstance.results.wasVoteCorrect(revealHash)) return VoteParticipation.CorrectVote;
       return VoteParticipation.WrongVote;
```

**Review by md0x — APPROVED** [review]
LGTM!

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol (hunk 7 above)
Comment (chrismaree) [reply packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:622]: good proposal! done

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol (hunk 6 above)
Comment (chrismaree) [reply packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:85]: yup! changed.

**Pull request merged**